modules/calendar/add_edit_event.php

```diff
@@ -1821,18 +1806,44 @@ function are_days_checked(){
 
 // Check for errors when the form is submitted.
 function validate(valu) {
-     var f = document.getElementById('theform');
+    var f = document.getElementById('theform');
     if ((f.form_repeat.checked || f.days_every_week.checked) &&
         (! f.form_enddate.value || f.form_enddate.value < f.form_date.value)) {
         alert('<?php echo addslashes(xl("An end date later than the start date is required for repeated events!")); ?>');
         return false;
     }
+
     //Make sure if days_every_week is checked that at least one weekday is checked.
     if(f.days_every_week.checked && !are_days_checked()){
         alert('<?php echo xls("Must choose at least one day!"); ?>');
         return false;
     }
 
+    //check if user selected appt. time is within clinic hours
```

Comment:
Wish we were keeping system time for everything 24 hour and only displaying AM/PM stuff...and only when an idiot specifically goes out of their way to demand that the system use that.  "Do you really want to use AM/PM?"  Do you Really REally want to do this?"  "OK, if you really really really want to do this, please sign the waiver form below in bright red ink, and return it to us before the ink turns rust-colored".

-anyone care to build the time.inc.php library?